service/migration_manager.cc

```diff
@@ -1094,7 +1095,10 @@ future<schema_ptr> migration_manager::get_schema_for_write(table_schema_version
         // Schema is synchronized through Raft, so perform a group 0 read barrier.
         // Batch the barriers so we don't invoke them redundantly.
         mlogger.trace("Performing raft read barrier because schema is not synced, version: {}", v);
-        co_await _group0_barrier.trigger(as);
+        utils::composite_abort_source composite_as;
```

Comment:
The abort sources are used inconsistently after this change.

I mean the `_as.abort_requested()` check at the beginning of this function; if we decide to compose the abort sources, shouldn't it be done at the beginning of the function, and all the checks refer to `composite_as`?

Reply:
Yeah, that was in particular the eventual race condition I was referring to - only checked at the beginning, but not being checked later.

Looking through the history, the function was much simpler initially and the check at the beginning made sense, now it should probably be moved. I still kept the check, but moved it into the `get_schema_definition()` method that is being called from here (as we probably still want to skip that part if there is an abort request); which will in particular also be using the composite source.